kernel/interrupts/src/aarch64/mod.rs

```diff
@@ -86,6 +91,15 @@ extern "C" fn default_irq_handler(exc: &ExceptionContext) -> EoiBehaviour {
     loop {}
 }
 
+#[inline(always)]
+fn read_timer_period_femtoseconds() -> u64 {
+    let counter_freq_hz = CNTFRQ_EL0.get();
+    let fs_in_one_sec = 1_000_000_000_000_000;
+    let period_femtoseconds = fs_in_one_sec / counter_freq_hz;
+    TICK_PERIOD_FEMTOSECS.call_once(|| period_femtoseconds);
+    period_femtoseconds
+}
```

Comment:
this is an unusual way to store a cached value. Typically you would do something like this:
```rust
fn read_timer_period_femtoseconds() -> u64 { 
    static TICK_PERIOD_FEMTOSECS: Once<u64> = Once::new();
    *TICK_PERIOD_FEMTOSECS.call_once(|| {
        let counter_freq_hz = CNTFRQ_EL0.get();
        let fs_in_one_sec = 1_000_000_000_000_000;
        let period_femtoseconds = fs_in_one_sec / counter_freq_hz;
        period_femtoseconds
    })
}
```
This has several advantages:
* the `static` variable is scoped within this function, ensuring nobody else can mess with it
* This function doesn't have to be called in an earlier init() function
* This function cannot fail -- if it hasn't been initialized, it initializes itself
* This also avoids the need to call `Once::try()`, which itself avoids the need to handle errors
* Guarantees the cached value will always be used, thanks to the semantics of `call_once`

Reply:
Ah yes sorry about that